.gitignore

```diff
@@ -52,6 +52,13 @@ lib/*.jar
 lib/native
 lib/ruby/cext/
 lib/ruby/gems/shared/gems
+lib/gems/*
+!lib/gems/gems/
+lib/gems/gems/*
+!lib/gems/gems/did_you_mean-1.0.0
+!lib/gems/specifications
+lib/gems/specifications/*
+!lib/gems/specifications/default
```

Comment:
At least in my local version of git `2.7.1` 
If you do `lib/gems`, it never evens looks inside this folder to find the unexclude:
`!lib/gems/gems/did_you_mean-1.0.0`

so if you do `lib/gems/**` your not excluding the `lib/gem` folder just all the files/folders inside of it so it can see/find the unexclude on the next line.

Please correct me if I am wrong though because I was hoping to do something that simple